### ISSUE TITLE
swipeLeft and swipeTop methods aren't working

### ISSUE THREAD

**Issue by sagunpandey** [alexbrillant/react-native-deck-swiper#187]
I stored a reference to the swiper.

```
ref={swiper => {
    this.swiper = swiper;
}}
```

However, only method swipeRight is working. Using swipeLeft and swipeTop throws our exception `TypeError: TypeError: TypeError: TypeError: undefined is not an object (evaluating '_this.swiper.swipeTop')`

![image](https://user-images.githubusercontent.com/3306020/46923907-870fef00-cfec-11e8-8e40-33b469767498.png)

**Comment by webraptor** [comment]
What's the code in MatchSwiper / index.js around line 37?

**Comment by sagunpandey** [comment]
```
onBookmarkClicked = () => {
    this.swiper.swipeTop(); // This is line 37
  };
```
`this.swiper.swipeRight()` is working though. Besides that, neither `swipeLeft()`, `swipeTop()` or `swipeBottom()` method works. It throws out exactly the same exeption as shown above.

And this is how I have been setting my reference to Swiper,

```
<Swiper
          ref={swiper => {
            this.swiper = swiper;
          }}
          cards={cards}
          cardIndex={cardIndex}
          renderCard={cardData => <MatchCard {...cardData} />}
          animateCardOpacity
          showSecondCard
          stackSize={cards.length}
          disableBottomSwipe
          cardVerticalMargin={10}
          cardHorizontalMargin={10}
          cardStyle={styles.card}
          onSwipedRight={this.onSwipedRight}
          onSwipedLeft={this.onSwipedLeft}
          onSwipedTop={this.onSwipedTop}
          {...props}>
</Swiper>
```

**Comment by sagunpandey** [comment]
> What's the code in MatchSwiper / index.js around line 37?

```
onBookmarkClicked = () => {
    this.swiper.swipeTop(); // This is line 37
  };
```
`this.swiper.swipeRight()` is working though. Besides that, neither `swipeLeft()`, `swipeTop()` or `swipeBottom()` method works. It throws out exactly the same exeption as shown above.

And this is how I have been setting my reference to Swiper,

```
<Swiper
          ref={swiper => {
            this.swiper = swiper;
          }}
          cards={cards}
          cardIndex={cardIndex}
          renderCard={cardData => <MatchCard {...cardData} />}
          animateCardOpacity
          showSecondCard
          stackSize={cards.length}
          disableBottomSwipe
          cardVerticalMargin={10}
          cardHorizontalMargin={10}
          cardStyle={styles.card}
          onSwipedRight={this.onSwipedRight}
          onSwipedLeft={this.onSwipedLeft}
          onSwipedTop={this.onSwipedTop}
          {...props}>
</Swiper>
```

**Comment by webraptor** [comment]
If you put a debugger within _onBookmarkClicked_ before the _swipeTop_ line what do you get on _this.swiper_ object?

**Comment by sagunpandey** [comment]
> If you put a debugger within _onBookmarkClicked_ before the _swipeTop_ line what do you get on _this.swiper_ object?

Luckily I have found the issue. I was setting onPress listener on my button as:

`<Button name="bookmark" onPress={this.onBookmarkClicked()} />`

I replaced this.onBookmarkClicked() by this.onBookmarkClicked and everything is working. It was invoking the method instead of just providing the reference to it, how silly of me. 

Thank you so much for your kind help.
